### PR TITLE
[Admin] Product Attributes View

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
|-----------------|-----
| Branch?         | 2.0 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no 
| License         | MIT

TODO:

- [x] Add Twig-Hooks configuration for all product attribute configurations
- [ ] Fix Update with PostgreSQL database set 

Result:

image
image

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductAttributesContext.php [hunk 1]
```diff
@@ -304,15 +304,15 @@ public function theAttributeWithCodeShouldNotAppearInTheStore(string $field, str
...
      */
-    public function iShouldSeeTheValue(string $value): void
+    public function iShouldSeeTheValueInLocale(string $value, string $localeCode): void
     {
         $content = $this->responseChecker->getResponseContent($this->client->getLastResponse());
         $choices = $content['configuration']['choices'];
 
         foreach ($choices as $values) {
-            if (in_array($value, $values)) {
+            if (in_array($value, $values) && isset($values[$localeCode])) {
```
Comment (Wojdylak) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductAttributesContext.php]:
```suggestion
            if ($values[$localeCode] === $value) {
```

File: src/Sylius/Bundle/AdminBundle/Form/Type/AttributeType/Configuration/SelectAttributeConfigurationType.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            'required' => false,
+            'entry_options' => [
+                'entry_type' => TextType::class,
+            ],
+        ]);
+    }
+
+    public function getBlockPrefix(): string
+    {
+        return 'sylius_attribute_type_configuration_select';
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Form/Type/AttributeType/Configuration/SelectAttributeConfigurationType.php]:
```suggestion
        return 'sylius_admin_attribute_type_configuration_select';
```

File: src/Sylius/Bundle/AdminBundle/Form/Type/AttributeType/Configuration/SelectAttributeChoicesCollectionType.php
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Form/Type/AttributeType/Configuration/SelectAttributeChoicesCollectionType.php:1]: Add `getBlockPrefix` method with the `sylius_admin_*` prefix

File: src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/product_attribute/update.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                template: '@SyliusAdmin/product_attribute/form/details/type.html.twig'
+            position:
+                template: '@SyliusAdmin/product_attribute/form/details/position.html.twig'
+            translatable:
+                template: '@SyliusAdmin/product_attribute/form/details/translatable.html.twig'
+
+        'sylius_admin.product_attribute.update.content.form.translations':
+            name:
+                template: '@SyliusAdmin/product_attribute/form/translations/name.html.twig'
+
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/product_attribute/update.yaml]:
```suggestion
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/services/form.xml [hunk 4]
```diff
@@ -273,5 +273,11 @@
         <service id="sylius_admin.form.type.zone" class="Sylius\Bundle\AdminBundle\Form\Type\ZoneType" >
             <tag name="form.type" />
         </service>
+
+        <service id="sylius.attribute_type.select" class="Sylius\Component\Attribute\AttributeType\SelectAttributeType">
+            <tag name="sylius.attribute.type" attribute-type="select" label="Select"
+                 form-type="Sylius\Bundle\AttributeBundle\Form\Type\AttributeType\SelectAttributeType"
+                 configuration-form-type="Sylius\Bundle\AdminBundle\Form\Type\AttributeType\Configuration\SelectAttributeConfigurationType" />
+        </service>
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/config/services/form.xml]: As this service is not a form type, I would be for moving it into another XML file

File: src/Sylius/Bundle/UiBundle/Resources/translations/messages.en.yml [hunk 5]
```diff
@@ -680,6 +680,7 @@ sylius:
         product_association_types: 'Product association types'
         product_associations: 'Product associations'
         product_attributes: 'Product attributes'
+        product_attribute: 'Product attribute'
```
Comment (GSadee) [inline src/Sylius/Bundle/UiBundle/Resources/translations/messages.en.yml]:
```suggestion
        product_attribute: 'Product attribute'
        product_attributes: 'Product attributes'
```

File: src/Sylius/Bundle/UserBundle/Tests/DependencyInjection/ConfigurationTest.php [hunk 6]
```diff
@@ -33,7 +33,7 @@ public function it_has_default_configuration_for_user_resetting(): void
                         'resetting' => [
                             'token' => [
                                 'ttl' => 'P1D',
-                                'length' => 16,
+                                'length' => 64,
```
Comment (GSadee) [inline src/Sylius/Bundle/UserBundle/Tests/DependencyInjection/ConfigurationTest.php]: It shouldn't be changed in this PR

Comment (Rafikooo) [reply src/Sylius/Bundle/UserBundle/Tests/DependencyInjection/ConfigurationTest.php]: I'll remove this commit once this PR is merged:
https://github.com/Sylius/Sylius/pull/16419

File: src/Sylius/Behat/Page/Admin/ProductAttribute/CreatePage.php
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/ProductAttribute/CreatePage.php:1]: Remove the entire class as it is empty

**Review by GSadee — APPROVED** [review]

**Pull request merged**